test/unit/lifecycle/property.js

```diff
@@ -323,5 +323,107 @@ describe('lifecycle/property', function () {
         create(opts);
       });
     });
+
+    describe('change()', function () {
+      it('is not called if the property has no default value', function () {
+        let calls = 0;
+        create({
+          change () {
+            ++calls;
+          }
+        });
+        expect(calls).to.equal(0);
+      });
+
+      it('is called if the property has no default value and it is set', function () {
+        let calls = 0;
+        let elem = create({
+          change () {
+            ++calls;
+          }
+        });
+        elem.test = true;
+        expect(calls).to.equal(1);
+      });
+
+      it('is not called if the property has a default value', function () {
+        let calls = 0;
+        create({
+          default: true,
+          change () {
+            ++calls;
+          }
+        });
+        expect(calls).to.equal(0);
+      });
+
+      it('is not called if the property has a default value and it does not change', function () {
+        let calls = 0;
+        let elem = create({
+          default: true,
+          change () {
+            ++calls;
+          }
+        });
+        elem.test = true;
+        expect(calls).to.equal(0);
+      });
+
+      it('is called if the property has a default value and it does change', function () {
+        let calls = 0;
+        let elem = create({
+          default: true,
+          change () {
+            ++calls;
+          }
+        });
+        elem.test = false;
+        expect(calls).to.equal(1);
+      });
+
+      it('is not called if the property value does not change', function () {
+        let calls = 0;
+        let elem = create({
+          default: true,
+          change () {
+            ++calls;
+          }
+        });
+        elem.test = true;
+        elem.test = true;
```

Comment:
That's to test whether or not `change()` is called if the value being set again is the same.